packages/qdrant-loader/src/qdrant_loader/core/state/checkpoint_manager.py

```
@@ -0,0 +1,284 @@
+                    source=source,
+                )
+                return None
+        except Exception as e:
+            logger.error(
+                "Failed to retrieve checkpoint",
+                project_id=project_id,
+                source_type=source_type,
+                source=source,
+                error=str(e),
+                error_type=type(e).__name__,
+            )
```

Ruff (0.15.15)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)